controllers/trainingjob/trainingjob_controller.go

```diff
@@ -308,6 +309,9 @@ func (r *Reconciler) updateStatusWithAdditional(ctx reconcileRequestContext, tra
 	ctx.Log.Info("updateStatusWithAdditional", "trainingJobPrimaryStatus", trainingJobPrimaryStatus, "trainingJobSecondaryStatus", trainingJobSecondaryStatus, "additional", additional)
 
 	jobStatus := &ctx.TrainingJob.Status
+
+	jobStatus.SageMakerTrainingJobName = ctx.TrainingJobName
```

Comment:
This feels unnecessarily complicated. We can just do the `Update` where we change the name (once per job).